services/brig/src/Brig/Whitelist.hs

```diff
@@ -1,44 +1,69 @@
 -- | > docs/reference/user/activation.md {#RefActivationWhitelist}
 --
 -- Email/phone whitelist.
-module Brig.Whitelist (Whitelist (..), verify) where
+module Brig.Whitelist
+    ( Whitelist (..)
+    , InternalWhitelist (..)
+    , verifyInternal
+    , verifyService
+    ) where
 
 import Imports
 import Data.Aeson
+import Data.Aeson.TH (deriveJSON)
 import Bilge.IO
 import Bilge.Request
 import Bilge.Response
 import Bilge.Retry
 import Brig.Types
 import Control.Monad.Catch (MonadMask, throwM)
 import Control.Retry
-import Data.Text
 import Data.Text.Encoding (encodeUtf8)
 import Network.HTTP.Client (HttpException (..), HttpExceptionContent (..), parseRequest)
+import Util.Options.Common (toOptionFieldName)
+
+import qualified Data.Text as T
 
 -- | A service providing a whitelist of allowed email addresses and phone numbers
+-- DEPRECATED: use internal whitelisting instead!
 data Whitelist = Whitelist
     { whitelistUrl  :: !Text     -- ^ Service URL
-    , whitelistUser :: !Text     -- ^ Username
-    , whitelistPass :: !Text     -- ^ Password
+    , whitelistUser :: !Text     -- ^ Service Username (basic auth)
+    , whitelistPass :: !Text     -- ^ Service Password (basic auth)
+
     } deriving (Show, Generic)
 
 instance FromJSON Whitelist
 
+data InternalWhitelist = InternalWhitelist
+    { internalWhitelistDomains :: ![Text]} deriving (Show, Generic)
+
+deriveJSON toOptionFieldName 'internalWhitelistDomains
+
+-- | Check internal whitelist for given email/phone number. Currently this
+-- consults a statically configured whitelist of allowed email domains..
+verifyInternal :: InternalWhitelist -> Either Email Phone -> Bool
+verifyInternal (InternalWhitelist domains) key =
+  let allowFromDomains (Left e) = (emailDomain e) `elem` domains
+      allowFromDomains _        = False
+  in allowFromDomains key
```

Comment:
so far, brig uses 4-space indentation everywhere (just like galley/cannon/proxy/cargohold - some code in `spar` has been 2-space indented, but at least so far it's consistent within the same file). We can of course have a debate on this, but generally I'd suggest keeping the indentation level of files as they are when modifying them (and not selectively indent some functions differently - that makes working with those files later much harder)

Reply:
Done.